engine/table/src/main/java/io/deephaven/engine/table/impl/remote/ConstructSnapshot.java

```diff
@@ -1509,61 +1531,184 @@ private static boolean serializeAllTable(
 
         final String[] columnSources = table.getDefinition().getColumnNamesArray();
 
-        try (final SharedContext sharedContext =
-                (columnSources.length > 1) ? SharedContext.makeSharedContext() : null) {
-            for (int ii = 0; ii < columnSources.length; ++ii) {
-                if (concurrentAttemptInconsistent()) {
-                    if (log.isDebugEnabled()) {
-                        final LogEntry logEntry = log.debug().append(System.identityHashCode(logIdentityObject))
-                                .append(" Bad snapshot before column ").append(ii);
-                        appendConcurrentAttemptClockInfo(logEntry);
-                        logEntry.endl();
-                    }
+        // Snapshot empty columns serially, and collect indices of non-empty columns
+        final int numColumnsToSnapshot =
+                (columnsToSnapshot != null) ? columnsToSnapshot.cardinality() : columnSources.length;
+        final TIntList nonEmptyColumnsIndices = new TIntArrayList(numColumnsToSnapshot);
+        if (!snapshotEmptyColumns(columnSources, columnsToSnapshot, table, logIdentityObject, snapshot,
+                nonEmptyColumnsIndices)) {
+            return false;
+        }
+        if (!nonEmptyColumnsIndices.isEmpty()) {
+            final ExecutionContext executionContext = ExecutionContext.getContext();
+            final boolean canParallelize =
+                    ENABLE_PARALLEL_SNAPSHOT &&
+                            executionContext.getOperationInitializer().canParallelize() &&
+                            nonEmptyColumnsIndices.size() > 1 &&
+                            (snapshot.rowsIncluded.size() > MINIMUM_PARALLEL_SNAPSHOT_ROWS ||
```

Comment:
I believe you want `>=` since minimum is inclusive.

Reply:
I saw that the check for `QueryTable.MINIMUM_PARALLEL_SELECT_ROWS` also had a `>` check, so kept the same. Do you think I should change it?